hy/compiler.py

```diff
@@ -366,6 +366,17 @@ def is_unpack(kind, x):
             and x[0] == "unpack_" + kind)
 
 
+# Provide asty.Foo(x, ...) as shorthand for
+# ast.Foo(..., lineno=x.start_line, col_offset=x.start_column)
+class Asty(object):
```

Comment:
...uhh, is this really necessary for this PR? I know it makes things a bit shorter, but it's also only used a few times and seems rather off-topic...